xrpl/asyncio/clients/websocket_base.py

```diff
@@ -18,19 +18,18 @@
 _REQ_ID_MAX: Final[int] = 1_000_000
 
 
-def _inject_request_id(request: Request) -> Request:
+def _inject_request_id(request: Dict[str, Any]) -> Dict[str, Any]:
     """
-    Given a Request with an ID, return the same Request.
+    Given a request dictionary with an ID, return the same dictionary.
 
-    Given a Request without an ID, make a copy with a randomly generated ID.
+    Given a request dictionary without an ID, make a copy with a randomly generated ID.
     """
-    if request.id is not None:
+    if "id" in request and request["id"] is not None:
         return request
-    request_dict = request.to_dict()
-    request_dict["id"] = f"{request.method}_{randrange(_REQ_ID_MAX)}"
-    resp = Request.from_dict(request_dict)
-    assert resp.id is not None  # mypy
-    return resp
+    request_dict = cast(Dict[str, Any], json.loads(json.dumps(request)))  # deepcopy
+    command = request["command"]
+    request_dict["id"] = f"{command}_{randrange(_REQ_ID_MAX)}"
```

Comment:
Could this ever fail by chance? (i.e. rolling the same id as a previous transaction?)

Reply:
Possibly, but the chance is very rare.

Reply:
I think that issue would be really hard to debug though (and the odds would increase as more requests were sent) Maybe something like this would be better? 

```
...
_REQ_ID_MAX = 1000000
last_id = 0
...
last_id = (1 + last_id) % _REQ_ID_MAX
request_dict["id"] = f"{command}_{last_id}"
```

Although I think that would mess up in a multi-thread situation unless locks were added, so maybe just doing random is better.

Reply:
Happy to chat more about that in a separate issue, but this is out of scope for this PR.